evfl/timeline.py

```diff
@@ -254,6 +254,7 @@ def _set_indexes_from_values(self) -> None:
             t.clip.set_index(clip_to_idx)
 
     def _do_write(self, stream: WriteStream) -> None:
+        self.triggers.sort(key=lambda a: a.clip.v.start_time + (a.type-1) * a.clip.v.duration)
```

Comment:
IMO it would be preferable (for readability reasons) to add a "get_trigger_time" function similar to this one:

https://github.com/open-ead/EventFlow/blob/c35d21b34397bec6dd3c67a649fc66b68839341e/src/evfl/TimelineObj.cpp#L77-L86

And then the sort just becomes `self.triggers.sort(key=lambda a: a.get_trigger_type())`.